transports/quic/tests/stream_compliance.rs

```diff
@@ -0,0 +1,78 @@
+use futures::channel::oneshot;
+use futures::StreamExt;
+use libp2p_core::Transport;
+use libp2p_quic as quic;
+use std::time::Duration;
+
+#[async_std::test]
+async fn close_implies_flush() {
+    let (alice, bob) = connected_peers().await;
+
+    libp2p_muxer_test_harness::close_implies_flush(alice, bob).await;
+}
+
+#[async_std::test]
+#[ignore] // Hangs forever, same as yamux. We can't read from a stream that we have never written to.
```

Comment:
This is not QUIC protocol specific, but implementation specific, right?

Reply:
@thomaseizinger isn't this one can be tricked with sending PING and waiting for PONG?

Reply:
Not quite, this one tests that we can read from a stream that we have never written to. Both `yamux` and QUIC seem to fail at that but I think in both cases, it is an implementation problem.

In reality, this isn't much of an issue in `libp2p` today because we always run multistream-select on the stream before we use it for the protocol. However, this also affects the TTFB metric and it would be great if we at some point have a muxer or a mechanism where we can open a stream that is already dedicated to a protocol. As such, I think these tests are worth having and we should fix the implementation eventually so they pass this test. The identify protocol f.e. opens a stream and requires the other party to just send data without us ever writing to the stream.

Reply:
> However, this also affects the TTFB metric and it would be great if we at some point have a muxer or a mechanism where we can open a stream that is already dedicated to a protocol.

Note that with [`V1Lazy`](https://docs.rs/multistream-select/0.10.2/multistream_select/enum.Version.html#variant.V1Lazy) and only a single stream protocol, multistream-select does not have an impact on TTFB.

That said, agreed. We should fix this.

Reply:
> > However, this also affects the TTFB metric and it would be great if we at some point have a muxer or a mechanism where we can open a stream that is already dedicated to a protocol.
> 
> Note that with [`V1Lazy`](https://docs.rs/multistream-select/0.10.2/multistream_select/enum.Version.html#variant.V1Lazy) and only a single stream protocol, multistream-select does not have an impact on TTFB.
> 
> That said, agreed. We should fix this.

It does for the listener right? In this particular testcase, the dialer doesn't write _any_ bytes to the stream and expects the listener to answer with the payload of a particular protocol right away.

Reply:
If we create a new stream, the other side won't be notified of the new stream until we actually start sending data. See quinn-rs/quinn#1425. Imo this makes sense since the initiator has to send some data on the stream first anyway so that the remote knows which application layer protocol belongs to (like it is done today with multistream-select, as @thomaseizinger already described above). 

> if we at some point have a muxer or a mechanism where we can open a stream that is already dedicated to a protocol.

Given that such a muxer is not planned for the near future I don't think this is something we have to fix or test now. 
The idea that one side simply sends an empty STREAM frame and with this triggers the remote to start sending a bunch of data on that stream generally feels a bit weird. Happy to be convinced otherwise :)
But I think that's out of scope for this PR, thus I'd be in favor of removing this test.

Reply:
Happy to remove it from the test suite. Will do in a separate PR!

Reply:
Sorry, I probably wasn't clear with what I meant. I don't think the test needs to be removed completely from the test-suite, since we may use it in the future to test a muxer that implements what you described above. I was just suggestion to remove this test within the QUIC transport since we will most likely not support it here.

Reply:
I don't think there is any point in having a test in the test suite that we don't end up running for every muxer.